service/src/main/java/org/cbioportal/genome_nexus/service/cached/CachedMyVariantInfoFetcher.java

```diff
@@ -1,25 +1,142 @@
-                                         MyVariantInfoDataFetcher fetcher)
+    public CachedMyVariantInfoFetcher(
+        ExternalResourceTransformer<MyVariantInfo> transformer,
+        MyVariantInfoRepository repository,
+        MyVariantInfoDataFetcher fetcher,
+        @Value("${myvariantinfo.max_page_size:200}") Integer maxPageSize,
+        ExternalResourceObjectMapper objectMapper
+    )
     {
-        super(MyVariantInfoRepositoryImpl.COLLECTION,
+        super(
+            MyVariantInfoRepositoryImpl.COLLECTION,
             repository,
             MyVariantInfo.class,
             fetcher,
-            transformer);
+            transformer,
+            maxPageSize
+        );
+
+        this.objectMapper = objectMapper;
+    }
+
+    protected Object buildRequestBody(Set<String> ids)
+    {
+        MultiValueMap<String, String> requestBody= new LinkedMultiValueMap<>();
+        requestBody.add("ids", Strings.join(ids, ','));
+        // TODO this way it looks cleaner, but it only fetches one result
+        // requestBody.put("ids", new ArrayList<>(ids));
+
+        return requestBody;
+    }
+
+    @Override
+    protected String extractId(MyVariantInfo instance)
+    {
+        return instance.getVariant();
+    }
+
+    @Override
+    protected String extractId(DBObject dbObject)
+    {
+        return (String)dbObject.get("_id");
+    }
+
+    @Override
+    protected DBObject normalizeResponse(DBObject rawJson)
+    {
+        List<DBObject> curatedList = new ArrayList<>();
+
+        // MyVariantInfo returns dynamic model (like error responses or inconsistent data model)
+        // which we cannot map exactly to our static model.
+        // Remove/normalize problematic ones so that we don't save those into the database
+        for (DBObject dbObject: Transformer.convertToDbObjectList(rawJson))
+        {
+            try {
+                // check for any errors (this throws exception if the response is an error JSON)
+                this.checkForErrorResponse(dbObject);
+
+                // rcv field seems inconsistent, convert to list if it is a plain object
+                this.normalizeRcv(dbObject);
+
+                // convert DBObject to a proper instance of the given class type
+                this.objectMapper.convertValue(dbObject, this.type);
+
+                // if can be successfully converted add to the curated list
+                curatedList.add(dbObject);
+            } catch (Exception e) {
+                LOG.warn(this.getId(dbObject) + ": " + e.getLocalizedMessage());
+            }
+        }
+
+        BasicDBList list = new BasicDBList();
+        list.addAll(curatedList);
+        return list;
+    }
+
+    private Object getId(@NotNull DBObject dbObject) {
+        return dbObject.get("_id") != null ? dbObject.get("_id"): dbObject.get("query");
+    }
+
+    private void normalizeRcv(@NotNull DBObject dbObject)
```

Comment:
I think I found a better/generic way to do this: https://stackoverflow.com/a/36245241

@inodb @leexgh I think we should make this a global setting for all other services we use. What do you think?

Reply:
Yup that sounds good 👍